test/policies/serializationPolicyTests.ts

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+import * as assert from "assert";
+import { HttpMethod } from "../../lib/httpMethod";
+import { HttpRequest } from "../../lib/httpRequest";
+import { HttpResponse } from "../../lib/httpResponse";
+import { InMemoryHttpResponse } from "../../lib/inMemoryHttpResponse";
+import { serializationPolicy } from "../../lib/policies/serializationPolicy";
+import { RequestPolicy } from "../../lib/requestPolicy";
+import { RequestPolicyFactory } from "../../lib/requestPolicyFactory";
+import { RequestPolicyOptions } from "../../lib/requestPolicyOptions";
+import booleanSpec from "../../lib/serialization/booleanSpec";
+import { compositeSpec } from "../../lib/serialization/compositeSpec";
+import dateSpec from "../../lib/serialization/dateSpec";
+import dateTimeRfc1123Spec from "../../lib/serialization/dateTimeRfc1123Spec";
+import numberSpec from "../../lib/serialization/numberSpec";
+import objectSpec from "../../lib/serialization/objectSpec";
+import sequenceSpec from "../../lib/serialization/sequenceSpec";
+
+describe("serializationPolicy", () => {
+  it("serializes request and response bodies", async () => {
+    const policyFactory: RequestPolicyFactory = serializationPolicy();
+
+    const inMemoryEchoServer: RequestPolicy = {
+      send: (request: HttpRequest) => {
+        return Promise.resolve(new InMemoryHttpResponse(request, 200, request.headers, JSON.stringify(request.serializedBody)));
+      }
+    };
+
+    const policy: RequestPolicy = policyFactory(inMemoryEchoServer, new RequestPolicyOptions());
+
+    const request = new HttpRequest(
```

Comment:
Consider modifying this constructor to take an object (allows for labeled arguments by convention, and passing a subset of optional arguments without having to `foo, null, null, bar`). e.g.

```ts
new HttpRequest({
  method: HttpMethod.GET,
  url: ...,
  headers: ...,
  body: ...,
  operationSpec: ...
})
```

Reply:
I agree. I'll do that.

Reply:
Done.